### PR TITLE
DOCS-8742-sockets-examples-editorial-kt

### PR DESCRIPTION
**Pull request by kevintroller**
Examples .adoc  -- editorial of **Create a Custom TCP Protocol** example

### CODE REVIEW

File: sockets/1.2/modules/ROOT/pages/edits/sockets-connector-examples.adoc [hunk 1]
```diff
@@ -0,0 +1,263 @@
...
+		<sockets:listener doc:name="Listener" config-ref="Sockets_Listener_config1" outputMimeType="application/json"/>
+		<sockets:send-and-receive doc:name="Send and receive" config-ref="Sockets_Request_config" outputMimeType="application/json">
+		</sockets:send-and-receive>
+		<logger level="INFO" doc:name="Logger" message='#[payload]'/>
+		<file:write path="musicians.json" config-ref="File_Config" mode="CREATE_NEW" />
+	</flow>
+----
+
+To test this example, use the same database setup as used for the sending example, run the Mule application, access the endpoint using the following netcat command `nc localhost 8085`, and send the data `{"id":2}`.
+
```
Comment (kbostrom) [inline sockets/1.2/modules/ROOT/pages/edits/sockets-connector-examples.adoc]: Edit the following example only.

File: sockets/1.2/modules/ROOT/pages/edits/sockets-connector-examples.adoc [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+== Create a Custom TCP Protocol
+
+The following example creates a TCP-based custom protocol that appends a string header to every message sent, and limits input message size to a fixed length by truncating all additional characters.
```
Comment (kbostrom) [inline sockets/1.2/modules/ROOT/pages/edits/sockets-connector-examples.adoc]:
```suggestion
The following example creates a TCP-based custom protocol that appends a string header to every message sent and limits input message size to a fixed length by truncating all additional characters.
```
Is this a common scenario?
Probably a silly question; I just kind of wonder how examples are chosen and if we should specify under what circumstances a user might encounter the need for this particular example.

File: sockets/1.2/modules/ROOT/pages/edits/sockets-connector-examples.adoc [hunk 3]
```diff
@@ -0,0 +1,263 @@
...
+
+. Create a class that implements the TcpProtocol interface:
```
Comment (kbostrom) [inline sockets/1.2/modules/ROOT/pages/edits/sockets-connector-examples.adoc]:
```suggestion
. Create a class that implements the `TcpProtocol` interface:
```

File: sockets/1.2/modules/ROOT/pages/edits/sockets-connector-examples.adoc [hunk 4]
```diff
@@ -0,0 +1,263 @@
...
+    try (BufferedOutputStream writer = new BufferedOutputStream(os)) {
+      writer.write(messageToSend, 0, HEADER.length() + dataLength);
+      writer.flush();
+    }
+  }
+}
+----
+
+[start=2]
+. Add the class name in the *Connector configuration* of the Socket operation:
```
Comment (kbostrom) [inline sockets/1.2/modules/ROOT/pages/edits/sockets-connector-examples.adoc]: The format following this step is sort of odd; in fact, the organization here is a bit different?

1. Step One, with long syntax example
2. Step Two, with two options and a syntax example that...applies to the second option? applies to both options?

I sort of wonder if this is a two-step example task or two separate subtasks....

Comment (kbostrom) [reply sockets/1.2/modules/ROOT/pages/edits/sockets-connector-examples.adoc]: Is this more accurate?
```suggestion
. Add the class name in the *Connector configuration* field of either the socket listener operation or one of the socket send operations:
```

Comment (kbostrom) [reply sockets/1.2/modules/ROOT/pages/edits/sockets-connector-examples.adoc]: ditto all the previous questions about capitalization in this three-PR set

File: sockets/1.2/modules/ROOT/pages/edits/sockets-connector-examples.adoc [hunk 5]
```diff
@@ -0,0 +1,263 @@
...
+
+* Socket Listener Operation +
+.. In Studio, open the *Global Element Properties* configuration for your Socket Listener operation.
+.. On the *General* tab, set the *Protocol* to `Custom protocol`.
```
Comment (kbostrom) [inline sockets/1.2/modules/ROOT/pages/edits/sockets-connector-examples.adoc]:
```suggestion
.. On the *General* tab, set *Protocol* to `Custom protocol`.
```

File: sockets/1.2/modules/ROOT/pages/edits/sockets-connector-examples.adoc [hunk 6]
```diff
@@ -0,0 +1,263 @@
...
+.. Add the class name in the *Protocol Class Name* field.
```
Comment (kbostrom) [inline sockets/1.2/modules/ROOT/pages/edits/sockets-connector-examples.adoc]:
```suggestion
.. Add the class name in the *Protocol Class Name* field:
```

File: sockets/1.2/modules/ROOT/pages/edits/sockets-connector-examples.adoc [hunk 7]
```diff
@@ -0,0 +1,263 @@
...
+
+.Socket Listener Custom Protocol Configuration
+image::socket-custom1-configuration.png[Socket Custom Protocol Configuration]
+
+* Socket Send, or Send and receive Operations +
```
Comment (kbostrom) [inline sockets/1.2/modules/ROOT/pages/edits/sockets-connector-examples.adoc]:
```suggestion
* Socket send operation or socket send and receive operation +
```

File: sockets/1.2/modules/ROOT/pages/edits/sockets-connector-examples.adoc [hunk 8]
```diff
@@ -0,0 +1,263 @@
...
+.. In Studio, open the *Global Element Properties* configuration of your Socket Send or Send and receive operations.
```
Comment (kbostrom) [inline sockets/1.2/modules/ROOT/pages/edits/sockets-connector-examples.adoc]:
```suggestion
.. In Studio, open the *Global Element Properties* configuration of your Socket Send operation or your socket Send and receive operations.
```

File: sockets/1.2/modules/ROOT/pages/edits/sockets-connector-examples.adoc [hunk 9]
```diff
@@ -0,0 +1,263 @@
...
+.. On the *Connection* tab, set the *Protocol* to `Custom protocol`.
```
Comment (kbostrom) [inline sockets/1.2/modules/ROOT/pages/edits/sockets-connector-examples.adoc]:
```suggestion
.. On the *Connection* tab, set *Protocol* to `Custom protocol`.
```

File: sockets/1.2/modules/ROOT/pages/edits/sockets-connector-examples.adoc [hunk 10]
```diff
@@ -0,0 +1,263 @@
...
+.. Add the class name in the *Protocol Class Name* field.
```
Comment (kbostrom) [inline sockets/1.2/modules/ROOT/pages/edits/sockets-connector-examples.adoc]:
```suggestion
.. Add the class name in the *Protocol Class Name* field:
```

**Review by kbostrom — DISMISSED** [review]
Edited per scope of request

**Review by dhirsch1 — APPROVED** [review]

**Pull request merged**